src/neural/factory.cc

```diff
@@ -123,7 +123,10 @@ std::unique_ptr<Network> NetworkFactory::LoadNetwork(
   } else {
     CERR << "Loading weights file from: " << net_path;
   }
-  const WeightsFile weights = LoadWeightsFromFile(net_path);
+  std::optional<WeightsFile> weights;
+  if (!net_path.empty()) {
```

Comment:
This means random backend needs an explicit --weights= since the default is auto discover.
But I'm fine with that...

Reply:
Err nevermind, I can't read.